src/database.py

```diff
@@ -0,0 +1,16 @@
+from sqlalchemy.ext.asyncio import create_async_engine, AsyncSession
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from src.settings import settings
+
+
+SQLALCHEMY_DATABASE_URL = settings.DATABASE_URL
+
+engine = create_async_engine(
+    SQLALCHEMY_DATABASE_URL, connect_args={"check_same_thread": False}
```

Comment:
The `connect_args={"check_same_thread": False}` is typically used with SQLite databases to allow connections to be shared across threads. Ensure this is necessary for your database setup, as it might not be applicable for other database systems.